File: capacity-club-api/src/domain-modules/workout/model/entity/workout.entity.ts
import { UniqueId, uniqueIdTransformer } from '@common/model/unique-id';
import { Transform } from 'class-transformer';
import { ApiProperty } from '@nestjs/swagger';
import { HomeTraining } from 'domain-modules/home-training/model/entity/home-training.entity';
import { IndoorTraining } from 'domain-modules/indoor-training/model/entity/indoor-training.entity';
import { TrainingCircuit } from 'domain-modules/training-circuit/model/entity/training-circuit.entity';
import { Entity, PrimaryColumn, Column, OneToMany, ManyToOne } from 'typeorm';
import { BaseEntity } from '@common/model';

/**
 * Entity representing a workout.
 * A workout includes multiple training circuits.
 */
@Entity()
export class Workout extends BaseEntity {
  /**
   * Unique identifier for the workout.
   * Uses a custom transformer for serialization.
   */
  @PrimaryColumn('varchar')
  @Transform(uniqueIdTransformer.to, { toClassOnly: true })
  @Transform(uniqueIdTransformer.from, { toPlainOnly: true })
  workout_id: UniqueId;
  @Column({ nullable: false })
  title: string;

  @OneToMany(
    () => TrainingCircuit,
    (trainingCircuit) => trainingCircuit.workout,
    {
      cascade: true,
      lazy: true,
    },
  )
  @ApiProperty({ type: () => TrainingCircuit, isArray: true })
  training_circuits: Promise<TrainingCircuit[]>;

  @ManyToOne(() => IndoorTraining, { nullable: true, lazy: true })
  @ApiProperty({ type: () => IndoorTraining })
  indoor_training: Promise<IndoorTraining>;

  @ManyToOne(() => HomeTraining, { nullable: true, lazy: true })
  @ApiProperty({ type: () => HomeTraining })
  home_training: Promise<HomeTraining>;
}

File: capacity_club_mobile_app/lib/domain-features/domain/workout/entity/workout_entity.dart
import 'package:capacity_club_mobile_app/core/model/entities/unique_id.dart';
import 'package:capacity_club_mobile_app/domain-features/domain/training-circuit/entity/training_circuit_entity.dart';

class WorkoutEntity {
  final UniqueId workout_id;
  final String title;
  final List<TrainingCircuitEntity> training_circuits;
  final bool is_empty;

  WorkoutEntity(
      {required this.workout_id,
      required this.title,
      required this.training_circuits,
      required this.is_empty});

  factory WorkoutEntity.empty() {
    print('ENTER WORKOUT ENTITY EMPTY');
    return WorkoutEntity(
        workout_id: UniqueId(''),
        title: '',
        training_circuits: [],
        is_empty: true);
  }
}
